CONTRIBUTING.rst

```diff
@@ -0,0 +1,161 @@
+Contributing
+============
+
+Instructions for contributors
+-----------------------------
+
+
+In order to make a clone of the GitHub_ repo: open the link and press the
+"Fork" button on the upper-right menu of the web page.
+
+I hope everybody knows how to work with git and github nowadays :)
+
+Workflow is pretty straightforward:
+
+  1. Clone the GitHub_ repo
+
+  2. Make a change
+
+  3. Make sure all tests passed
+
+  4. Add a file into ``CHANGES`` folder (`Changelog update`_).
+
+  5. Commit changes to own ray clone
+
+  6. Make pull request from github page for your clone against master branch
+
+  .. note::
+     If your PR has long history or many commits
+     please rebase it from main repo before creating PR.
+
+Preconditions for running ray test suite
+--------------------------------------------
+
+We expect you to use a python virtual environment to run our tests.
+
+There are several ways to make a virtual environment.
+
+If you like to use *virtualenv* please run:
+
+.. code-block:: shell
+
+   $ cd ray
+   $ virtualenv --python=`which python3` venv
+   $ . venv/bin/activate
+
+For standard python *venv*:
+
+.. code-block:: shell
+
+   $ cd ray
+   $ python3 -m venv venv
+   $ . venv/bin/activate
+
+For *virtualenvwrapper*:
+
+.. code-block:: shell
+
+   $ cd ray
+   $ mkvirtualenv --python=`which python3` ray
+
+There are other tools like *pyvenv* but you know the rule of thumb
+now: create a python3 virtual environment and activate it.
+
+After that please install libraries required for development:
+
+.. code-block:: shell
+
+   $ pip install -r requirements/dev.txt
+
+.. note::
+  If you plan to use ``pdb`` or ``ipdb`` within the test suite, execute:
+
+.. code-block:: shell
+
+    $ py.test tests -s
+
+  command to run the tests with disabled output capturing.
+
+Congratulations, you are ready to run the test suite!
+
+
+Run ray test suite
+----------------------
+
+After all the preconditions are met you can run tests typing the next
+command:
+
+.. code-block:: shell
+
+   $ python test/runtest.py
+
+The command at first will run the *flake8* tool (sorry, we don't accept
+pull requests with pep8 or pyflakes errors).
+
+On *flake8* success the tests will be run.
+
+Please take a look on the produced output.
+
+Any extra texts (print statements and so on) should be removed.
+
+
+Documentation
+-------------
+
+We encourage documentation improvements.
+
+Please before making a Pull Request about documentation changes run:
+
+.. code-block:: shell
+
+   $ pip install -r requirements-doc.txt
+   $ make html
+
+Once it finishes it will output the index html page
+``open file:///.../ray/docs/_build/html/index.html``.
+
+Go to the link and make sure your doc changes looks good.
+
+
+Changelog update
```

Comment:
This section isn't relevant, is it?

Reply:
If you send a PR about improving the docs I think it's a great idea the check the docs (so open the local link and...).

About Changelog, how do you collect the changes so far?
This is really useful if you want to collect info between releases and you know what changed.

Reply:
We currently don't maintain a changelog. We could certainly add one, though we can always construct it after the fact just by looking through the commit history.

Reply:
Ok, so I will remove this changelog part.